### ISSUE TITLE
how to deal with helm tiller pod of CrashLoopBackOff   state?

### ISSUE THREAD

**Issue by lixiaobing1** [kubernetes/helm#2322]
root@ubuntu:/home/zte/share/helm/linux-amd64# kubectl get pods --namespace kube-system
NAME                             READY     STATUS             RESTARTS   AGE
tiller-deploy-1332173772-2jjbd   0/1       CrashLoopBackOff   521        2d
root@ubuntu:/home/zte/share/helm/linux-amd64# kubectl logs --namespace kube-system tiller-deploy-1332173772-2jjbd 
root@ubuntu:/home/zte/share/helm/linux-amd64# kubectl --namespace kube-system describe tiller-deploy-1332173772-2jjbd
the server doesn't have a resource type "tiller-deploy-1332173772-2jjbd"
root@ubuntu:/home/zte/share/helm/linux-amd64# 

2017-04-23 21:42:37.154155 W | etcdserver: apply entries took too long [252.427668ms for 1 entries]
2017-04-23 21:42:37.154193 W | etcdserver: avoid queries with large range/delete range!
I0423 21:42:49.835915    7193 kuberuntime_manager.go:458] Container {Name:tiller Image:gcr.io/kubernetes-helm/tiller:v2.3.1 Command:[] Args:[] WorkingDir: Ports:[{Name:tiller HostPort:0 ContainerPort:44134 Protocol:TCP HostIP:}] EnvFrom:[] Env:[{Name:TILLER_NAMESPACE Value:kube-system ValueFrom:nil}] Resources:{Limits:map[] Requests:map[]} VolumeMounts:[] LivenessProbe:&Probe{Handler:Handler{Exec:nil,HTTPGet:&HTTPGetAction{Path:/liveness,Port:44135,Host:,Scheme:HTTP,HTTPHeaders:[],},TCPSocket:nil,},InitialDelaySeconds:1,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} ReadinessProbe:&Probe{Handler:Handler{Exec:nil,HTTPGet:&HTTPGetAction{Path:/readiness,Port:44135,Host:,Scheme:HTTP,HTTPHeaders:[],},TCPSocket:nil,},InitialDelaySeconds:1,TimeoutSeconds:1,PeriodSeconds:10,SuccessThreshold:1,FailureThreshold:3,} Lifecycle:nil TerminationMessagePath:/dev/termination-log TerminationMessagePolicy:File ImagePullPolicy:IfNotPresent SecurityContext:nil Stdin:false StdinOnce:false TTY:false} is dead, but RestartPolicy says that we should restart it.
I0423 21:42:49.836131    7193 kuberuntime_manager.go:742] checking backoff for container "tiller" in pod "tiller-deploy-1332173772-2jjbd_kube-system(0d832ef7-2663-11e7-a8db-4c09b4b0c553)"
I0423 21:42:49.836319    7193 kuberuntime_manager.go:752] Back-off 5m0s restarting failed container=tiller pod=tiller-deploy-1332173772-2jjbd_kube-system(0d832ef7-2663-11e7-a8db-4c09b4b0c553)
E0423 21:42:49.836370    7193 pod_workers.go:182] Error syncing pod 0d832ef7-2663-11e7-a8db-4c09b4b0c553 ("tiller-deploy-1332173772-2jjbd_kube-system(0d832ef7-2663-11e7-a8db-4c09b4b0c553)"), skipping: failed to "StartContainer" for "tiller" with CrashLoopBackOff: "Back-off 5m0s restarting failed container=tiller pod=tiller-deploy-1332173772-2jjbd_kube-system(0d832ef7-2663-11e7-a8db-4c09b4b0c553)"

**Comment by michelleN** [comment]
hey @lixiaobing1 
Would love to help here but I'm unsure of how we got to this state. Would you mind giving me some more information? 
What version of Helm and Tiller are you running? What version of Kubernetes are you running? Did you have a working Tiller which later went into this state or did it never come up in the first place?

Thank you.

**Comment by tricky42** [comment]
Hi,

it looks like I am running into the same problems. I installed Minikube on my Macbook and started a fresh k8s v1.6 cluster. Then I installed Tiller by calling
>helm init

This finishes successfully and I see in the K8S dashboards the Tiller deployments, replication set and pods appearing.

When I now try to install a chart (e.g. I tried MongoDB and Redis) by executing 
>helm install --debug --name mongo stable/mongodb
Created tunnel using local port: '60743'
SERVER: "localhost:60743"
Fetched stable/mongodb to mongodb-0.4.0.tgz
CHART PATH: /Users/d061245/hybris/play/tyk_k8s/mytyk/mongodb-0.4.0.tgz
Error: transport is closing

this operation fails. The log output of the Tiller pods looks like this:

>kubectl -n kube-system logs tiller-deploy-747649721-9wqrq
Tiller is listening on :44134
Probes server is listening on :44135
Storage driver is ConfigMap
2017-05-18 16:24:27.880086 I | storage.go:133: Getting release history for 'mongo'
2017-05-18 16:24:27.889732 I | release_server.go:873: Executing pre-install hooks for mongo
2017-05-18 16:24:27.889760 I | release_server.go:900: Hooks complete for pre-install mongo
2017-05-18 16:24:27.889764 I | storage.go:133: Getting release history for 'mongo'
error: group map[policy:0xc4203c80e0 :0xc4203c9e30 authentication.k8s.io:0xc4203c9f10 autoscaling:0xc4204620e0 certificates.k8s.io:0xc420462310 componentconfig:0xc4200460e0 extensions:0xc4203c8070 rbac.authorization.k8s.io:0xc4203c8150 storage.k8s.io:0xc4203c81c0 apps:0xc4203c9ea0 authorization.k8s.io:0xc420462070 batch:0xc4204622a0 federation:0xc4203c83f0] is already registered

>kubectl -n kube-system get pods -o wide
NAME                            READY     STATUS    RESTARTS   AGE       IP               NODE
kube-addon-manager-minikube     1/1       Running   1          22m       192.168.99.100   minikube
kube-dns-268032401-rt8jj        3/3       Running   1          22m       172.17.0.4       minikube
kubernetes-dashboard-kjv41      1/1       Running   1          22m       172.17.0.3       minikube
tiller-deploy-747649721-9wqrq   1/1       Running   3          21m       172.17.0.2       minikube

Any idea what is causing the crashes? 

K8S:         v1.6.3
Minikube: v0.19.0
Kubectl:    v1.6.3
